sr_utilities_common/scripts/sr_utilities_common/performance_test.py

```diff
@@ -0,0 +1,109 @@
+
+        self._set_control_loop_values(diagnostics_control_loop_msg)
+        if self._hand_h:
+            self._set_ethercat_values(diagnostics_ethercat_msg_list)
+
+    def _ethercat_data_callback(self, msg):
+        msg_list = list()
+        msg_list.append(msg)
+        self._set_ethercat_values(msg_list)
+
+    def _set_ethercat_values(self, diagnostics_ethercat_msg_list):
+        invalid_packets_total_count_list = list()
+        for diagnostics_ethercat_msg in diagnostics_ethercat_msg_list:
+            if self._hand_h:
+                for value_list in diagnostics_ethercat_msg.values:
+                    if value_list.key == "Invalid Packets Total":
+                        invalid_packets_total_count_list.append(float(value_list.value))
+                    if value_list.key == "Invalid Packets Recent":
+                        self.total_invalid_packets_recent += float(value_list.value)
+                self.total_invalid_packets_total_count = deepcopy(invalid_packets_total_count_list)
+            elif self._hand_e:
+                if diagnostics_ethercat_msg.motor_data_type.data == 0:
+                    self._invalid_hand_e_packets_total += 1
+
+    def _set_control_loop_values(self, diagnostics_control_loop_msg):
+        for value_list in diagnostics_control_loop_msg.values:
+            if value_list.key == "Control Loop Overruns":
+                control_loop_overruns_count = value_list.value
+            if value_list.key == "Recent Control Loop Overruns":
+                recent_control_loop_overruns_count = value_list.value
+
+        self.total_control_loop_overruns_count = float(control_loop_overruns_count)
+        self.total_recent_control_loop_overruns_count += float(recent_control_loop_overruns_count)
+
+    def _print_results(self, duration):
+        avg_recent_control_loop_overruns_count = self.total_recent_control_loop_overruns_count / duration
+        avg_total_invalid_packets_recent = self.total_invalid_packets_recent / duration
+
+        rospy.loginfo("Control Loop Overruns: {}".format(self.total_control_loop_overruns_count))
+        rospy.loginfo("Recent Control Loop Overruns: {}".format(avg_recent_control_loop_overruns_count))
+        if self._hand_h:
+            for i, value in enumerate(self.total_invalid_packets_total_count):
+                rospy.loginfo("Invalid Packets Total in finger {}: {}".format(i, value))
+            rospy.loginfo("Invalid Packets Recent: {}".format(avg_total_invalid_packets_recent))
+        else:
+            rospy.loginfo("Invalid Packets Total: {}".format(self._invalid_hand_e_packets_total))
+
+    def run(self, duration=20.0):
+        if not self._hand_h and not self._hand_e:
+            rospy.logerr("Not recognize hand!")
+            return False
+        start_time = rospy.get_time()
+        time = 0.0
+        while time < duration and not rospy.is_shutdown():
+            time = rospy.get_time() - start_time
+            rospy.sleep(0.0001)
```

Comment:
Please use a more sensible sleep value, like 0.1 or 0.5
Remember this loop has nothing to do with the reception of the messages, it only checks the time to end the script, so it doesn't need to spin that fast.